### ISSUE TITLE
Creating a new resource at the 'Global' level sets the user group as 'prefs'

### ISSUE THREAD

**Issue by grantfitzsimmons** [specify/specify7#3956]
Probably related to #3918, #3828, and #3780

When creating a new app resource at the 'Global' level, it is currently added to the User Group `Prefs` instead of `Common` as it was before!

### Correct Behavior:
`v7.8.0`:
<img width="720" alt="image" src="https://github.com/specify/specify7/assets/37256050/fe8f0dc3-2654-4273-9703-c5b0f47051ed">

### Incorrect Behavior:
`v7.8.13`:
<img width="664" alt="image" src="https://github.com/specify/specify7/assets/37256050/58579f63-77ab-49e8-8d4a-ff32ed8efa9a">

This should be resolved and investigated ASAP.

**Comment by grantfitzsimmons** [comment]
<img width="1016" alt="image" src="https://github.com/specify/specify7/assets/37256050/d0d06059-3858-4f92-bd56-8bf4c7acfdde">

Happened again on the `insectcollection-umn` deployment. 🤦‍♂️ 

We _need_ to resolve this!

**Comment by CarolineDenis** [comment]
@grantfitzsimmons , can you test is this behavior is still happening on xml-editor branch please?

**Comment by grantfitzsimmons** [comment]
Same behavior is happening on the `xml-editor` branch here as well

**Comment by CarolineDenis** [comment]
NOTES: 
-  Global Preferences and Preferences are created during database initialization
-  When opening for the 1st time Sp7, Global Preferences and Preferences are present
-  Users do not have access to Sp App Resource Directory, hence are not able to modify the User Group field 
-  Creating a new global-level app resource automatically assigns it to the Common User Group
-  To access the Sp App Resource Directory, comment out SpAppResourceDir in SpAppResource: autoGenerateViewDefinition() in webOnlyViews.ts.
-  Sp allows only one Global Pref and one Preferences setting
- 
ISSUES: 
- Deleting and recreating Global Preferences or Preferences results in Sp assigning the User Group as Common.
- Some databases lack the common directory, causing new resources to be assigned to Prefs instead of Common.

**Comment by maxpatiiuk** [comment]
it looks like back-end is expecting user type to be "Common" for the global level resources:
https://github.com/specify/specify7/blob/5f7808f91f3afe2eca57506dbec482c35818880b/specifyweb/context/app_resource.py#L153
given this, it appears that we should either:
- make back-end look for global resources in all directories (by making it ignore user type when searching at global level)
- OR make front-end create a global resource with type Common, even if other global resources already exist

Then, Prefs and Global Prefs are special cases because those are explicitly looked at in directories with specific user types: https://github.com/specify/specify7/blob/fix-dynamic-pref-titles/specifyweb/context/remote_prefs.py#L6-L20
  - Not something front-end should handle as we aren't creating remote prefs and global prefs on the front end (if anything, we are deprecating these prefs files) - when it comes to fetching, since these files are created by sp6, we can assume that they are in a correct directory already

@grantfitzsimmons are you aware if user type for global resources matters in sp6 beyond the cases outlined above?

**Comment by CarolineDenis** [comment]
@maxpatiiuk for "Not something front-end should handle as we aren't creating remote prefs and global prefs on the front end (if anything, we are deprecating these prefs files) - when it comes to fetching, since these files are created by sp6, we can assume that they are in a correct directory already" 

he issue arises when the user deletes the preferences ('prefs' or 'global prefs') in Sp7 and needs to recreate them. In such cases, they are assigned to 'Common' instead of 'Prefs' or 'Global Prefs'. Should we disable the delete button for those 2 resources?

**Comment by grantfitzsimmons** [comment]
> The issue arises when the user deletes the preferences ('prefs' or 'global prefs') in Sp7 and needs to recreate them. In such cases, they are assigned to 'Common' instead of 'Prefs' or 'Global Prefs'. Should we disable the delete button for those 2 resources?

# Yes.

We need to make sure these two resources are pre-populated (they are not, unless someone used Specify 6 first), and we need to disable the **creation** or **deletion** for both of these.

**Comment by maxpatiiuk** [comment]
I was under the assumption that these are created at a database creation (which happens in sp6) - I might be wrong.
I haven't yet seen a single database that doesn't have these (unless someone manually deleted them?).
And once again, can we talk more about implementing Institutional Prefs and deprecating Global Prefs and Prefs?

**Comment by CarolineDenis** [comment]
@maxpatiiuk , I think you're right about this " I haven't yet seen a single database that doesn't have these (unless someone manually deleted them?)."

**Comment by grantfitzsimmons** [comment]
@maxpatiiuk @CarolineDenis I researched the test panel databases, and it appears that's most often the case. It seems the data is typically only missing if it was manually deleted or replaced.

I couldn't ignore the fact that I recall that preferences were missing a few times. It was problematic to create a new record because of the same issues we are looking to resolve now.

From my research, it looks like Global Prefs were created sometime after Remote User Prefs in most databases.

For most databases, the `preferences` app resource are `SpAppResourceID` 1 and 2, but occasionally these were clearly removed and re-added by someone.

For some, particularly the newer databases, such as the `archireef` database we made in 7 and imported to 6 almost immediately, it only has **1 preference** file, which is Global Prefs. There is no Remote User Prefs unless someone used 6.

In the `geology` database (one I made in Specify 6 and uploaded to the test panel immediately after), only the Global Prefs existed. When I connected via Specify 6 it created the Remote User Prefs resource. This was 5 days after I created the database, further evidence that these are not _always_ present.

## Wizard Behavior

I created a new database with only the Botany discipline using the SpWizard application to test this theory.

1. Only one `preferences` app resource (Global Prefs) is created, containing the following:
 
   ```
   #Global Prefs 
   #Thu Dec 14 10:33:36 CST 2023. 
   GEONAMES_LAST_MOD=1677708482000 
   ```

2. I logged in as the user I created during the wizard process.
3. Immediately upon logging in, the Remote User Prefs `preferences` app resource is created.

   ```
   #Remote User Prefs 
   #Thu Dec 14 10:35:21 CST 2023 
   recent_collection_id.spadmin.preferences_testing=4 
   ```

---

**TL;DR:**

The wizard creates the `Global Prefs` app resource. The `Remote User Prefs` resource is only created if a user logs into the database using Specify 6. If the database is generated and only used with Specify 7, the `Remote User Prefs` file is never created.

**Comment by grantfitzsimmons** [comment]
>And once again, can we talk more about implementing Institutional Prefs and deprecating Global Prefs and Prefs?

This is the ideal solution.

**Comment by maxpatiiuk** [comment]
thanks. given the above information and the fact that I would rather not spend time fixing what we are about to throw out, I would suggest we do the following changes to fix the other related issues Grant linked:

> Probably related to https://github.com/specify/specify7/issues/3918, https://github.com/specify/specify7/issues/3828, and https://github.com/specify/specify7/issues/3780

I proposed two solutions in:
https://github.com/specify/specify7/issues/3956#issuecomment-1853251752:

> 1. make back-end look for global resources in all directories (by making it ignore user type when searching at global level)
> 2. OR make front-end create a global resource with type Common, even if other global resources already exist

not quite sure which is better so am fine with both.

@CarolineDenis how does that sound? if so, feel free to work on implementing this

side note: the user type used by sp6 is part of sp6 permission system - sp7 does not use user types anywhere anymore except for the app resources. so if possible, we should not deepen our involvement with user types.
(instead of user types, sp7 has roles - which we should consider integrating deeper into app resources and other parts, especially once we are more separated from sp6)

**Comment by realVinayak** [comment]
@maxpatiiuk something like this? 

This change ignores usertype for  'Common' directory: https://github.com/specify/specify7/commit/7c7a212a4cae9f443fbf1f8f2b84895d37e80444

This change makes ExportFeed explicitly for 'Common' directory (so, if created as 'prefs', it will still be found) https://github.com/specify/specify7/commit/59e6a9152d89adab3d1a4b442269bcb3cc4befde
